lib/core/collection/array.nit

```diff
@@ -475,8 +475,8 @@ class Array[E]
 		var l = length
 		if l != o.length then return false
 		var i = 0
-		var it = _items
-		var oit = o._items
+		var it = _items.as(not null)
+		var oit = o._items.as(not null)
 		while i < l do
```

Comment:
the case where the length is 0 may crash.
Add a `if l == 0 then return true` after the `if l != o.length then return false`